sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -245,6 +246,55 @@ case class DataSourceResolution(
       catalog: TableCatalog,
       identifier: Identifier,
       create: CreateTableStatement): CreateV2Table = {
+
+    val isCaseSensitive = conf.caseSensitiveAnalysis
+    // Check that columns are not duplicated in the schema
+    val flattenedSchema = SchemaUtils.explodeNestedFieldNames(create.tableSchema)
+    SchemaUtils.checkColumnNameDuplication(
```

Comment:
This is not allowed for any table, right? I think it should be added to CheckAnalysis.

Reply:
Sure. Old code paths didn't check nested fields though (which is a bug)